src/access/wac.test.ts

```diff
@@ -0,0 +1,383 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it } from "@jest/globals";
+import { IriString, SolidDataset, WithServerResourceInfo } from "../interfaces";
+import { dataset } from "../rdfjs";
+import { getAgentAccess } from "./wac";
+import { Response } from "cross-fetch";
+import { triplesToTurtle } from "../formats/turtle";
+import { mock_addAclRuleQuads } from "../acl/mock.internal";
+import { acl, foaf } from "../constants";
+
+jest.mock("../fetcher.ts", () => ({
+  fetch: jest.fn().mockImplementation(() =>
+    Promise.resolve(
+      new Response(undefined, {
+        headers: { Location: "https://arbitrary.pod/resource" },
+      })
+    )
+  ),
+}));
+
+function getMockDataset(
+  sourceIri: IriString,
+  aclIri?: IriString
+): SolidDataset & WithServerResourceInfo {
+  return Object.assign(dataset(), {
+    internal_resourceInfo: {
+      sourceIri: sourceIri,
+      isRawData: false,
+      linkedResources: {},
+      aclUrl: aclIri,
+    },
+  });
+}
+
```

Comment:
It's a bit annoying given that you've already done all the work of setting this up this way, but... Since this really builds on top of the existing WAC functions, which are independently tested already, I'm wondering if we should not just re-use that API in our tests, and also the mock functions defined in `acl/mock.ts`? As in, there's no need for us to verify here that it really does construct the right type and structure of the datasets, so testing that here means more code to keep up-to-date if that internal implementation ever changes, without it helping us catch other regressions that we wouldn't have caught otherwise.

Reply:
If I understand your suggestion correctly, that would mean doing things like 
```
it("fetches the resource ACL if available", async () => {
      
    let aclResource: AclDataset = Object.assign(getMockDataset("https://some.pod/resource.acl"), {
      internal_accessTo: "https://some.pod/resource"
    })
    aclResource = setAgentResourceAccess(aclResource, "https://some.pod/profile#agent", {
      read: true, append: false, write: false, control: false 
    });

    const mockFetch = jest.fn(window.fetch).mockResolvedValue(
      mockResponse(await triplesToTurtle(Array.from(aclResource)), {
        status: 200,
        url: "https://some.pod/resource.acl",
      })
    );

    const resource = getMockDataset(
      "https://some.pod/resource",
      "https://some.pod/resource.acl"
    );
    const result = getAgentAccess(resource, "https://some.pod/profile#agent", {
      fetch: mockFetch
    });

    await expect(result).resolves.toStrictEqual({
      read: true,
      append: undefined,
      write: undefined,
      controlRead: undefined,
      controlWrite: undefined,
    });
  });
```
If so, this was my initial approach, but `setAgentResourceAccess` and the like create blank nodes, which are usually resolved when saved in a remote dataset. However in this case since they are not (since we don't actually go and save them), and being blank nodes they aren't matched by `getAgentAccess` from `acl/agent`, which is the function we should use. Mocking the actual content of the ACL using the pre-existing mock function was easier.

However, I did replace the implementation of `getMockDataset ` with our own internal mocks, because I ironically had forgotten about them ^^

Reply:
See d7ae3fd3ff7fbd5782f5de373924b8db552c700b